### PR TITLE
Also parse async function statement in export node

### PR DESCRIPTION
**Pull request by XeCycle**
AsyncFunctionDeclaration is added as a
HoistableDeclaration (http://tc39.github.io/ecmascript-asyncawait/#HoistableDeclaration),
which shall be covered in productions "export Declaration" and "export
default HoistableDeclaration".

### CODE REVIEW

File: tests/parser.el [hunk 1]
```diff
@@ -799,6 +806,12 @@ the test."
 (js2-deftest-parse parse-export-generator-declaration "export default function* one() {\n}")
 (js2-deftest-parse parse-export-assignment-expression "export default a = b;")
 
+(js2-deftest-parse parse-export-async-function-allow-await
+  "export async function f() {\n  await f();\n}\n;")
```
Comment (XeCycle) [inline tests/parser.el]: Note here; the trailing `\n;` also appears in `export function f() {}`.

Comment (dgutov) [inline tests/parser.el]: Will you add a corresponding fix to `js2-print-export-node`?

Comment (XeCycle) [inline tests/parser.el]: Okay, will make that fix commit before this one.

**Comment by dgutov** [comment]
Thanks!

**Pull request merged**